lib/llm/providers/gemini.rb

```diff
@@ -28,15 +23,23 @@ def complete(prompt, params = {})
       auth req
       res = request @http, req
 
-      Response.new(
-        JSON.parse(res.body)["candidates"].map { |candidate|
-          Message.new(candidate.dig("content", "role"), candidate.dig("content", "parts", 0, "text"))
-        }
-      )
+      Response::Completion.new(res.body, self)
     end
 
     private
 
+    ##
+    # @param (see LLM::Provider#completion_messages)
+    # @return (see LLM::Provider#completion_messages)
+    def completion_messages(raw)
+      raw["candidates"].map do
+        LLM::Message.new(
+          _1.dig("content", "role"),
+          _1.dig("content", "parts", 0, "text")
```

Comment:
@antaz Should `Message#text` be an array ? In this context it can have "parts"

Reply:
@0x1eef I'm not sure what Gimini refers to by parts, yet